core/src/main/java/org/apache/iceberg/TableMetadataParser.java

```diff
@@ -77,8 +76,8 @@ public static void write(TableMetadata metadata, OutputFile outputFile) {
     }
   }
 
-  public static String getFileExtension(Configuration configuration) {
-    return ConfigProperties.shouldCompress(configuration) ? ".metadata.json.gz" : ".metadata.json";
+  public static String getFileExtension(boolean isCompressed) {
```

Comment:
What do you think about moving the compression extension to before `.metadata.json`? Then file format detection wouldn't need to worry about multiple extensions.

Reply:
Could you elaborate a bit more on this one?

There are other places that depend on having ".gz" at the end of the file name, but we should be able to update them as well.

Reply:
We would still have to guess the filename for valid compression codecs, so this doesn't help that problem. But putting the compression extension before the `metadata.json` extension would allow the detection code in `FileFormat` to work independent of compression.

I updated that for the metadata tables: https://github.com/apache/incubator-iceberg/pull/252/files#diff-38bc235aaa75fff2db21b4df58501da9R61

Reply:
That's a good idea. Do we need to update our Python implementation as well?

Reply:
I'm not sure if python supports compressed metadata yet. @TGooch44, there are some naming changes with file-based tables.